File: user-service/src/main/java/org/aston/ems/user_service/service/CustomDigitComparator.java
package org.aston.ems.user_service.service;


import org.aston.ems.user_service.dto.UserDTO;
import org.springframework.stereotype.Service;

import java.util.Comparator;

@Service
public class CustomDigitComparator implements Comparator<UserDTO> {
    @Override
    public int compare(UserDTO o1, UserDTO o2) {
        return Integer.compare(o1.getMark() - o2.getMark(), 0);
    }
}

File: user-service/src/main/java/org/aston/ems/user_service/service/CustomDigitComparator.kt
package org.aston.ems.user_service.service

import org.aston.ems.user_service.dto.UserDTO
import org.springframework.stereotype.Service


@Service
class CustomDigitComparator : Comparator<UserDTO> {
    override fun compare(o1: UserDTO, o2: UserDTO): Int {
        return 0.compareTo(o2.mark - o1.mark)
    }
}
